### PR TITLE
Add ratified Svadu extension

### PR DESCRIPTION
**Pull request by ved-rivos**
## Description

Add support for ratified Svadu extension.

### Related Issues
Sail - https://github.com/riscv/sail-riscv/pull/260

### Ratified/Unratified Extensions

- [X] Ratified
- [ ] Unratified

### List Extensions

https://github.com/riscv/riscv-svadu

### Reference Model Used

- [x ] SAIL
- [ x] Spike
- [ ] Other - < SPECIFY HERE >

### Mandatory Checklist:

  - [x ] All tests are compliant with the test-format spec present in this repo ?
  - [ x] Ran the new tests on RISCOF with SAIL/Spike as reference model successfully ?
  - [ ] Ran the new tests on RISCOF in [coverage mode](https://riscof.readthedocs.io/en/stable/commands.html#coverage)
  - [ ] Link to Google-Drive folder containing the new coverage reports ([See this](https://github.com/riscv-non-isa/riscv-arch-test/blob/main/CONTRIBUTION.md#uploading-test-stats) for more info): < SPECIFY HERE >
  - [ ] Link to PR in RISCV-ISAC from which the reports were generated : < SPECIFY HERE > 
  - [ ] Changelog entry created with a minor patch

### Optional Checklist:

  - [ ] RISCV-V CTG PR link if tests were generated using it : < SPECIFY HERE >
  - [ x] Were the tests hand-written/modified ?
  - [ ] Have you run these on any hard DUT model ? Please specify name and provide link if possible in the description
  - [ ] If you have modified arch\_test.h Please provide a detailed description of the changes in the Description section above.

### CODE REVIEW

File: riscv-test-suite/rv32i_m/svadu/svadu.S [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    RVTEST_CASE(1,"//check ISA:=regex(.*32.*); check ISA:=regex(.*I.*Zicsr.*); def rvtest_mtrap_routine=True; def TEST_CASE_1=True",sv32)
+
+    RVTEST_SIGBASE(x13, signature_x13_1)
+  	
+    # Turn on VM
+    li a0, (SATP_MODE & ~(SATP_MODE<<1)) * SATP_MODE_SV32
+    la a1, test_pte_sv32
+    srl a1, a1, RISCV_PGSHIFT
+    or a1, a1, a0
+    csrw sptbr, a1
```
Comment (allenjbaum) [inline riscv-test-suite/rv32i_m/svadu/svadu.S:31]: This was renamed SATP long ago

File: riscv-test-suite/rv32i_m/svadu/svadu.S [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    sfence.vma
+
+    # setup PTE
```
Comment (allenjbaum) [inline riscv-test-suite/rv32i_m/svadu/svadu.S:34]: I think this is better done before enabling VM

File: riscv-test-suite/rv32i_m/svadu/svadu.S (hunk 1 above)
Comment (ved-rivos) [reply riscv-test-suite/rv32i_m/svadu/svadu.S:31]: muscle memory :(
This file got included however by mistake. I had fixed it in the files I intended to push. I will pull this one out.

File: riscv-test-suite/rv32i_m/svadu/svadu.S (hunk 2 above)
Comment (ved-rivos) [reply riscv-test-suite/rv32i_m/svadu/svadu.S]: Please ignore this file. It was pushed by mistake. I have pulled it out of the PR now.

**Comment by ved-rivos** [comment]
The extension name is updated to Svhad now.

**Review by UmerShahidengr — APPROVED** [review]

**Comment by UmerShahidengr** [comment]
Merged this one, as its equivalent ctg PR has already been approved.

**Pull request merged**